crypto/provider_core.c

```diff
@@ -824,6 +825,43 @@ int OSSL_PROVIDER_get_conf_parameters(OSSL_PROVIDER *prov, OSSL_PARAM params[])
     return 1;
 }
 
+int OSSL_PROVIDER_conf_get_bool(const OSSL_PROVIDER *prov,
+                                const char *name, int defval)
+{
+    char *val = NULL;
+    OSSL_PARAM param[2] = { OSSL_PARAM_END, OSSL_PARAM_END };
+
+    param[0].key = (char *)name;
+    param[0].data_type = OSSL_PARAM_UTF8_PTR;
+    param[0].data = (void *) &val;
+    param[0].data_size = sizeof(val);
+    param[0].return_size = OSSL_PARAM_UNMODIFIED;
+
+    /* Errors are ignored, returning the default value */
+    if (OSSL_PROVIDER_get_conf_parameters(prov, param)
+        && OSSL_PARAM_modified(param)
+        && val != NULL) {
+        if ((strcmp(val, "1") == 0)
+            || (strcmp(val, "yes") == 0)
+            || (strcmp(val, "YES") == 0)
+            || (strcmp(val, "true") == 0)
+            || (strcmp(val, "TRUE") == 0)
+            || (strcmp(val, "on") == 0)
+            || (strcmp(val, "ON") == 0)) {
```

Comment:
Why not use OPENSSL_strcasecmp thrice instead?

Reply:
Is `OPENSSL_strcasecmp` available in the FIPS module?

Reply:
> Is `OPENSSL_strcasecmp` available in the FIPS module?

Yes.  Most things are available in the FIPS provider.